src/OrleansSQLUtils/CreateOrleansTables_MySql.sql

```diff
@@ -185,19 +184,13 @@ CREATE TABLE OrleansQuery
 		' 
 		INSERT INTO OrleansMembershipVersionTable
 		(
-			DeploymentId,
-			Version
+			DeploymentId
 		)
-		SELECT	
-			@deploymentId,
-			@version
-		WHERE NOT EXISTS
-		(			
-			SELECT 1
-			FROM OrleansMembershipVersionTable
-			WHERE DeploymentId = @deploymentId AND @deploymentId IS NOT NULL
-			FOR UPDATE
-		);
+		Values
+		(
+			@deploymentId
+		)
+		ON DUPLICATE KEY UPDATE DeploymentId = DeploymentId;
```

Comment:
I'm not that familiar with MySQL, but as there is no conditional, it looks like in case `@deploymentId` already exists, all the deployments in the table will be updated to this one given as a parameter. In case there is only this one, would it get updated without visible effect, but would it still produce 1 on `ROW_COUNT` call and hence make later logging deviate from what is for other storage backends.

As a side note to casual watcher: In the SQL Server version there is a conditional here to check if this succeeded or not. It stems from the original intent to treat Azure Table storage as the normative one and log if there was a table already. One can see them at https://github.com/dotnet/orleans/blob/a64fa8953e764ac391ebd549d395ac66374b545f/src/OrleansAzureUtils/Storage/AzureBasedMembershipTable.cs#L31 and at https://github.com/dotnet/orleans/blob/a64fa8953e764ac391ebd549d395ac66374b545f/src/OrleansSQLUtils/Messaging/SqlMembershipTable.cs#L32. If this logging deviates per storage, it may make troubleshooting (on public forums) more difficult.